typed-racket-lib/typed-racket/private/parse-type.rkt

```diff
@@ -55,6 +56,11 @@
 ;; current-term-names when parsing ψ
 (define current-local-term-ids (make-parameter '()))
 
+;; current-in-struct-prop: Paremeter<Boolean>
+;; This parameter is used to assure that some types
+;; are only allowed to used in struct properties
```

Comment:
This comment is not very clear. I get the idea, but please fix the grammar, use Racket syntax for the type, and maybe mention Self & Imp